### DISCUSSION TITLE
Pregnant problem

### DISCUSSION THREAD

**Discussion by Yuelia-Lunare** [ErinaSugino/Starbound-SBR-Mod#142] [Trouble Shooting] [answered]
Hello, recently I started sexbound reborn and I cleaned a lot of mods to make everything work

I managed to resolve some problem after a complete check, but there are still problems

Currently I'm trying to make my character pregnant, which seems to work, but once I teleport to my ship it's like the whole thing resets and I'm no longer pregnant ( and probably the other NPCs too)

I apologize in advance for the inconvenience, and especially if it's a stupid bug that I haven't seen

(I also have other bugs like 3-way sex which doesn't work, but I prefer to do one thing at a time in another topic)

[starbound.log](https://github.com/user-attachments/files/17607272/starbound.log)

**Comment by ErinaSugino** [comment]
The answer is quite simple:

```
[15:17:57.911] [Info] Current gender for #-65536 is male
[15:17:57.924] [Info] Entity -65536 cannot ovulate (anymore), removing pregnancies and ovulation cycle logic
```

How are you even getting a male entity pregnant?

EDIT:
Also, to go ahead and crunch down troubleshooting time.

```
[15:16:43.802] [Info] Root: Detected asset source named 'Lox_AnimatedTentaclePlant' at '..\mods\AnimatedTentaclePlant.pak'
[15:16:43.802] [Info] Root: Detected unnamed asset source at '..\mods\Lustbound Base Manager'
[15:16:43.802] [Info] Root: Detected unnamed asset source at '..\mods\MorePoses_0.1.1'
```
Get rid of these mods.

**Reply by Yuelia-Lunare** [reply]
I admit I don't really know, in the sexbound version I could be pregnant and "produce" children

I had to adjust this in the options so that the males were too (Or maybe I misunderstood the English) I'm talking about the reborn version

I will do some tests and screenshots for you the next day, for me it is 00:30 AM

The pregnancy test in any case showed that I had become pregnant, and once teleported I tried again, it showed me the opposite, I went back to the planet in question, I have the impression that the NPC is no longer pregnant and neither am I (usually there is a text above their head that indicates if they are pregnant)

EDIT : I just saw your edit, I'll do that tomorrow and give you an update!

**Reply by ErinaSugino** [reply]
Yes, that would be the observed behavior. Your character gets pregnant, the pregnancy data is stored and persists. Upon the next init of your entity, which happens upon teleport to another world server instance, the body trait check runs again, and deletes the pregnancy data. This is the same for both player and NPC, although NPCs reset quite more often - everytime the teleport around, or get unloaded because they are offscreen.

**Comment by Yuelia-Lunare** [comment]
Here I am again! 

I removed the 3 mods in question that you mentioned

so I started my problem again with screenshots, sending you the logs

Edit : Using a birthing pill (which accelerates pregnancy) works.
The problem would be to prevent this verification which removes the pregnancy so, what can I do? or make sure that males can be authorized

I listen to your solutions and your advice

[starbound.log](https://github.com/user-attachments/files/17609842/starbound.log)

![Capture d’écran (6586)](https://github.com/user-attachments/assets/cb8b0104-7971-4017-9242-f72c07812a4b)
![Capture d’écran (6589)](https://github.com/user-attachments/assets/2c8151a3-a2c9-48d0-97d5-4b87e3a47995)
![Capture d’écran (6590)](https://github.com/user-attachments/assets/215a13cd-91fb-4670-b68b-f8407aa16682)
![Capture d’écran (6591)](https://github.com/user-attachments/assets/092c1a63-43a3-4fb5-80b1-f9eccc2742ba)
![Capture d’écran (6592)](https://github.com/user-attachments/assets/49b69b88-1435-40ac-b505-78808db4387a)

**Reply by ErinaSugino** [reply]
None of this adds anything new. As mentioned, the behavior that males cannot have babies and hence are cleaned up on init or sub-gender change is working as normal. The question is why your male even gets pregnant in the first place.

Have you turned on "enableFreeForAll" or something in the config?

**Reply by Yuelia-Lunare** [reply]
yes exactly that's why he gets pregnant, as I said I put that in the options

**Reply by ErinaSugino** [reply, marked as answer]
Yeah, that's not accounted for entity sided.
You'd need to overwrite the main "male" gender in the subGenderList and set "canOvulate" to true.

**Reply by Yuelia-Lunare** [reply]
I'm sorry but I don't understand how to do it ><

I should overwrite what in which config?

The only "subgender" I found is this but I can't find anything like "Male" or "canOvulate"
I must have misunderstood something ><

![Capture d’écran (6604)](https://github.com/user-attachments/assets/2cc90067-0f4a-41de-ab2a-26de2163b63c)

**Reply by ErinaSugino** [reply]
Not the subgender plugin config, the main config. Has a list of defined sub-genders.

**Reply by Yuelia-Lunare** [reply]
Oh I see, I found it!
Sorry

I'm going to try this right away

Besides, deleting the mods that you told me fixed the 3-way sex, I won't have to bother you with another subject ^^ (my translation is not perfect sorry again)

**Reply by Yuelia-Lunare** [reply]
No it still doesn't work,  I don't understand what I'm doing wrong

[starbound.log](https://github.com/user-attachments/files/17611293/starbound.log)
![Capture d’écran (6605)](https://github.com/user-attachments/assets/32b095d1-f5b7-4ed3-849e-0919d293984f)

**Reply by ErinaSugino** [reply]
You... have renamed the setting to "derList". Obviously no sub-genders are working now cause the setting does not exist anymore.

**Reply by Yuelia-Lunare** [reply]
I put it back as before, the problem is that I don't see any other list

I don't understand where I should place it

      {
        "name": "male",
        "canOvulate": true,
        "canProduceSperm": true,
        "hasPenis": true,
        "hasVagina": false
      }

**Reply by ErinaSugino** [reply]
You put it in the right place, but you broke the entire thing by renaming the setting "subGenderList" to "derList"

**Reply by Yuelia-Lunare** [reply]
The thing is, I never renamed it...

I will therefore rename it as you say then

edit : actually looking in the github its correct, it's strange that I "derList" given that I don't even know what it could correspond to

**Reply by Yuelia-Lunare** [reply]
oh no I think I understood, when I wanted to search I clicked on something who has deleted, which is why I would never find the list you were saying...

oh I hate myself

**Reply by Yuelia-Lunare** [reply]
Good ! after correcting my lamentable error, it works now!

thank you for your patience ><
and thank you also for devoting your time to your mod and for your help
